
# Assembling Merged Reads to Generate Contigs

## 0. Introduction and Preparation

After the process of merging reads, the next step involves assembling these merged reads into contigs using Unicycler, a de novo assembler optimized for bacterial genomes.

**Required software and tools for the assembling process:**

- [Unicycler v0.5.0](https://github.com/rrwick/Unicycler): Unicycler combines the advantages of several assembly algorithms and strategies to generate high-quality sequence assemblies.

- [Python3](https://www.python.org/downloads/): Required to run Unicycler.

## 1. Input

The input for the assembly process consists of merged reads typically ending with the suffix `_merged.fq.gz`.

Change the current directory to the location where the merged files are stored:

```bash
cd /merged_files
```

## 2. Output

The output from the assembly process will be FASTA files with the suffix `.fasta`. Define the output directory for Unicycler:

```bash
UNICYCLER_OUTPUT_DIR='/unicycler_output'
```

## 3. Choice 1: Assembling Merged Reads

Unicycler is used to assemble merged single reads into contigs.

**-s** is used to indicate single-end read mode

**-o** specifies the output directory

### Executing the Assembly

Run Unicycler on all merged reads with the suffix `.fq.gz`, extracting the sample name from each input file to appropriately name the output directory.

```bash
for file in *.fq.gz; do
    base_name=$(basename "$file" .fq.gz)
    output_dir="$UNICYCLER_OUTPUT_DIR/${base_name}_unicycler_output"
    mkdir -p "$output_dir"
    unicycler -s "$file" -o "$output_dir"
    echo "Unicycler processing completed for $base_name"
done
```

## 4. Choice 2: Assembling Paired-End Reads

Unicycler is also capable of assembling paired-end reads: 

**-1** is one file of paired end reads

**-2** is another file of the same sample

```bash
unicycler -1 short_reads_1.fastq.gz -2 short_reads_2.fastq.gz -o output_dir
```

In this document, both choices for running Unicycler have been outlined: one for single-end (merged) reads and another for paired-end reads. Each section includes the necessary commands and options to efficiently run the assembly process.
