# Running the Nextflow Pipeline for Genome Processing

This document outlines the process of running a Nextflow pipeline that handles genome preprocessing, assembly, and annotation using a cluster environment.

## Run Pipeline

### Load Nextflow Module on the Cluster

First, load the Nextflow software module necessary for running the pipeline:

```bash
module load apps/nextflow
```
### Execute the Nextflow Script

Run the main Nextflow pipeline script using Slurm as the job scheduler:

```bash
nextflow run main.nf -profile slurm
```

### Specify Data and Output Directories

Define the directories for input data and pipeline outputs:

```
--datadir=/group/ecoli_acras/Conor/Pipeline_C3/1_Preprocessing/1.2_CuratedData/1.2_Quality_trimmed/ --outdir=/group/result
```

## 0. Preparation for Pipeline

Before starting, ensure that Nextflow is installed. 

- [Nextflow](https://www.nextflow.io/docs/latest/getstarted.html#installation)

## 1. Create the Pipeline Script (module.nf)

This section details the script `module.nf`, which contains all necessary processes for assembly and annotation.

### 1.1 Merge Process

The merging process combines paired-end reads into a single file, handling potential overlaps and unmatched reads.

```bash
process merge{
        label 'high_res'
        memory { (2.GB * task.attempt) }
        scratch 'true'
        publishDir "${params.outdir}" , mode: 'copy'

        input:
        tuple val(sampleId), file(reads)

        output:
        path ("*_merged.fq.gz"), emit: merged_results

        script:
        """
        /usr/share/bbmap/bbmerge.sh in1=${reads[0]} in2=${reads[1]} out=${sampleId}_merged.fq.gz outu=${sampleId}_unmerged.fq.gz ihist=${sampleId}.ihist.gz
        """
}
```

## 1.2 Assembly Process

Following merging, this script assembles the reads into contiguous sequences (contigs).

```bash
process assembly{
        label 'norm_res'
        memory { (2.GB * task.attempt) }
        scratch 'true'
        publishDir "${params.outdir}" , mode: 'copy'

        input:
        path merged_file

        output:
        path ("*_assembly.fasta"), emit: assembly_result

        script:
        """
        mkdir ${merged_file.baseName}_unicycler_output
        unicycler -s "$merged_file" -o ${merged_file.baseName}_unicycler_output
        cp -vf  ${merged_file.baseName}_unicycler_output/assembly.fasta ${merged_file.baseName}_assembly.fasta
        """
}
```
## 1.3 Annotation Process

The final process in the pipeline annotates the assembled genome sequences, identifying gene features and other elements.

```
process annotation{
        label 'low_res'
        memory { (2.GB * task.attempt) }
        scratch 'true'
        publishDir "${params.outdir}" , mode: 'copy'

        input:
        path assembly_file

        output:
        path ("*_annotation"), emit: annotation_result

        script:
        """
        mkdir ${assembly_file.baseName}_annotation
        prokka --force --proteins ${params.ecoli_data} --outdir ${assembly_file.baseName}_annotation --prefix ${assembly_file.baseName} ${assembly_file}
        """
}
```

