# Quality Assessment for Assembled and Annotated Genomes

This documentation explains how to assess the quality of genome assemblies using specialized tools. The process is divided into two choices based on different needs: comparing assemblies from different pipelines and assessing a large number of assemblies from the same pipeline.

## Choice 1: Quality Assessment for Assemblies to Compare Assemblies from Different Pipelines

### Required Software and Tools

- [**QUAST**](https://github.com/ablab/quast): QUAST (QUality ASsessment Tool) evaluates genome assemblies by computing various metrics like N50, NGA50, and the number of misassemblies.

### Input

Assembled genomes are expected in FASTA format, typically named `assembly.fasta` within each pipeline output directory.

```bash
NEW_ASSEMBLY_DIR=/new_assemblies/
EXIST_ASSEMBLY_DIR_2=/exist_assemblies/
```

### Output

Analysis reports are generated in the specified output directory.

```bash
QUAST_OUTPUT_DIR=/quast_output/
```

### Execution

Run QUAST to compare assemblies from different sources corresponding to the same sample name. Here, QUAST is run with a reference genome and GenBank file for detailed comparison.

**-r** means reference assembly

**-g** means reference genbank

**-o** means output

**-t 1** means single-thread

```bash
mkdir -p "$QUAST_OUTPUT_DIR"
for dir in "$NEW_ASSEMBLY_DIR"/*/; do
    sample_name=$(basename "$dir" | sed 's/_merged_unicycler_output//')
    new_assembly_file="${dir}assembly.fasta"
    exist_assembly_file="$EXIST_ASSEMBLY_DIR/${sample_name}.fasta"
    output_dir="$QUAST_OUTPUT_DIR/quast_results_${sample_name}"
        if [[ -f "$new_assembly_file" && -f "$exist_assembly_file" ]]; then
        quast.py -r "/path/to/reference/genome/ecoli_k12.fna" -g "/path/to/reference/genome/ecoli_k12.gbff" -o "$output_dir" "$new_assembly_file" "$exist_assembly_file" -t 1
    else
        [[ ! -f "$new_assembly_file" ]] && echo "New assembly .gfa file not found in $dir"
        [[ ! -f "$exist_assembly_file" ]] && echo "Exist assembly .fasta file not found for $sample_name"
    fi
done
```
## Chioce 2: Quality Assessment for a Large Number of Assemblies from the Same Pipeline

### Required Software and Tools

- [MetaQUAST](https://github.com/ablab/metaquast): Suitable for assessing multiple assemblies, MetaQUAST automates the evaluation process.

### Input

Input assemblies should be available in FASTA format within the designated directory.

### Output

Generates a comprehensive analysis report in the specified output directory.

### Execution

Run MetaQUAST to assess multiple assemblies simultaneously. This script is optimized for handling large datasets by processing all FASTA files in the given directory.

**-r** means reference assembly

**-g** means reference genbank

**-o** means output

**-t 1** means single-thread 

```bash
cd assemblies_folder
metaquast.py $(ls *.fasta) -r /ecoliK12.fna -g /ecoliK12.gff -o output -t 1
```